src/fastener.js

```diff
@@ -126,9 +126,17 @@ function downToU(key, z) {
   if (isArray(focus) && isNumber(key) && 0 <= key && key < focus.length)
     return fromArray(focus, key, dissocPartialU("focus", z))
 }
-
 export const downTo = curry2(downToU)
 
+function downPathU([k, ...ks], z) {
+  const { focus } = z
+  if ((isObject(focus) || isArray(focus)) && k in focus) {
+    const d = downToU(k, z)
+    return ks.length > 0 ? downPathU(ks, d) : d
+  }
+}
```

Comment:
Note that `downTo` (and the Uncurried version `downToU`) already check that the operation is valid and return `undefined` in case it isn't.  This way the `downPath` operation can be seen as a partial reduce.

Also, it is useful to allow paths to be empty.  This way paths can be treated as full monoids rather than as just semigroups.

```js
function downPathU(path, z) {
  for (let i=0, n=path.length; z && i<n; ++i)
    z = downPathU(path[i], z)
  return z
}
```

Hmm... Looking at this, it might be useful to make zipper movement operations to allow the zipper argument to be `undefined` and consistently map movements from `undefined` to `undefined`.